src/workerd/api/tests/js-rpc-test.js

```diff
@@ -1179,11 +1197,40 @@ export let crossContextSharingDoesntWork = {
 
 export let waitUntilWorks = {
   async test(controller, env, ctx) {
-    globalWaitUntilPromise = null;
-    await env.MyService.testWaitUntil();
+    // Tests ctx.waitUntil
+    {
+      globalWaitUntilPromise = null;
+      await env.MyService.testWaitUntil();
+
+      assert.strictEqual(globalWaitUntilPromise instanceof Promise, true);
+      await globalWaitUntilPromise;
+    }
+
+    // Tests `import { waitUntil } from 'cloudflare:workers` on DO
+    {
+      globalWaitUntilPromise = null;
+      await env.MyService.testImportedWaitUntil();
+
+      assert.strictEqual(globalWaitUntilPromise instanceof Promise, true);
+      await globalWaitUntilPromise;
+    }
 
-    assert.strictEqual(globalWaitUntilPromise instanceof Promise, true);
-    await globalWaitUntilPromise;
+    // Tests `import { waitUntil } from 'cloudflare:workers` directly
+    {
+      globalWaitUntilPromise = null;
+
+      let resolve;
+      globalWaitUntilPromise = new Promise((r) => {
+        resolve = r;
+      });
+
+      waitUntil(
+        (async () => {
+          await scheduler.wait(100);
+          resolve();
+        })()
+      );
```

Comment:
bit more compact and readable ...

```suggestion
      const { promise, resolve } = Promise.withResolvers();
      waitUntil(scheduler.wait(100).then(resolve));
      await promise;
```